README.md

```diff
@@ -115,7 +127,195 @@ or globally
 
 With global non-static class you can also reset `StaticClass` per file by setting the value to anything but `false`.
 
-### Linked resource file namespaces
+### Partial classes (per file or globally)
+
+To extend an existing class, you can make your classes partial.
+
+```xml
+<ItemGroup>
+  <EmbeddedResource Update="Resources\ArtistCategoriesNames.resx">
+    <PartialClass>true</PartialClass>
+  </EmbeddedResource>
+</ItemGroup>
+```
+
+or globally
+
+```xml
+<PropertyGroup>
+  <ResXFileCodeGenerator_PartialClass>true</ResXFileCodeGenerator_PartialClass>
+</PropertyGroup>
+```
+
+### Static Members (per file or globally)
+
+In some rare cases it might be useful for the members to be non-static.
+
+```xml
+<ItemGroup>
+  <EmbeddedResource Update="Resources\ArtistCategoriesNames.resx">
+    <StaticMembers>false</StaticMembers>
+  </EmbeddedResource>
+</ItemGroup>
+```
+
+or globally
+
+```xml
+<PropertyGroup>
+  <ResXFileCodeGenerator_StaticMembers>false</ResXFileCodeGenerator_StaticMembers>
+</PropertyGroup>
+```
+
+## Inner classes (per file or globally)
+
+If your resx files are organized along with code files, it can be quite useful to ensure that the resources are not accessible outside the specific class the resx file belong to.
+
+```xml
+<ItemGroup>
+	<EmbeddedResource Update="**/*.resx">
+		<DependentUpon>$([System.String]::Copy('%(FileName).cs'))</DependentUpon>
+		<InnerClassName>MyResources</InnerClassName>
+		<InnerClassVisibility>private</InnerClassVisibility>
+		<InnerClassInstanceName>EveryoneLikeMyNaming</InnerClassInstanceName>
+	    <StaticMembers>false</StaticMembers>
```

Comment:
You need to add these new options to [VocaDb.ResXFileCodeGenerator.props](https://github.com/VocaDB/ResXFileCodeGenerator/blob/main/VocaDb.ResXFileCodeGenerator/build/VocaDb.ResXFileCodeGenerator.props), so that these properties are [visible to the compiler](https://github.com/dotnet/roslyn/blob/main/docs/features/source-generators.cookbook.md#consume-msbuild-properties-and-metadata).

Reply:
Ahh cool, didn't even know that.